core/src/main/java/com/linecorp/armeria/common/logging/MetricLabel.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+/**
+ * User definable label name for metrics (ex. {@link io.prometheus.client.Collector}).
+ * @param <T> the implementing class
+ */
+public interface MetricLabel<T extends MetricLabel> extends Comparable<T> {
```

Comment:
I suggested this because I've found it convenient to be able to use an enum, which automatically implements this interface, for label-type of things. But String would probably be fine too.